lsp-mode.el

```diff
@@ -1867,34 +1870,42 @@ WORKSPACE is the workspace that contains the progress token."
                       (t (->> (eq :workspace lsp-diagnostics-modeline-scope)
                               (lsp-diagnostics)
                               (ht-values)
-                              (-flatten))))))
-    (->> diagnostics
-         (-group-by #'lsp:diagnostic-severity?)
-         (-sort (-lambda ((left) (right))
-                  (> right left)))
-         (-map (-juxt (-compose #'lsp--severity-code->severity #'cl-first)
-                      (-compose #'length #'cl-rest)))
-         (-map (-lambda ((code count))
-                 (propertize (format "%s" count)
-                             'face (cl-case code
-                                     ('error 'error)
-                                     ('warning 'warning)
-                                     ('info 'success)
-                                     ('hint 'success)))))
-         (s-join "/")
-         (format "%s"))))
+                              (-flatten-n 1))))))
+    (--> diagnostics
```

Comment:
the whole block till s-join could be rewritten with one -reduce. Without that, we will create 2 intermediate copies of the initial collection(one coming from -group-by and one from -map). Even the flatten above can be removed.

Reply:
I've rewritten it and also remove the sort.